config/config.go

```diff
@@ -367,7 +389,13 @@ func DiscoverFederation() error {
 
 	result, err := httpClient.Do(req)
 	if err != nil {
-		return errors.Wrapf(err, "Failure when doing federation metadata lookup to %s", discoveryUrl)
+		var netErr net.Error
+		if errors.As(err, &netErr) {
```

Comment:
This doesn't limit to only timeouts.  `*net.Error` appears to have a separate `Timeout` method.

```suggestion
		if errors.As(err, &netErr) && netErr.Timeout() {
```

By taking the underlying error object and turning it to a string, you loose the error hierarchy.  You really want:

```
MetadataTimeoutErr.Wrap(err)
```

and to add an `Unwrap` method.